docs/articles/reference/upgrading-to-v16.md

```diff
@@ -0,0 +1,258 @@
+## Upgrading to v0.16
+
+Bytewax v0.16 introduces major changes to the way that custom Bytewax inputs are created, as well as improvements to and windowing and execution. In this article, we'll cover some of the updates we've made to our API to make upgrading to v0.16 easier.
+
+## Input changes
+
+In v0.16, we have restructured input to be based around partitions in order to support rescaling stateful dataflows in the future. We have also dropped the `Config` suffix to our input classes. As an example, `KafkaInputConfig` has been renamed to `KafkaInput`.
+
+`ManualInputConfig` is now replaced by two base superclasses that can be subclassed to create custom input sources. They are `DynamicInput` and `PartitionedInput`.
+
+### `DynamicInput`
+
+`DynamicInput` sources are input sources that support reading distinct items from any number of workers concurrently.
+
+Dynamic inputs must be able to supply disjoint data for each worker that is started. If multiple workers read the same data, the Dataflow will process those as duplicate items.
+
+`DynamicInput` sources do not support storing resume state.
+
+### `PartitionedInput`
+
+`PartitionedInput` sources are input sources that manage a fixed number of independent partitions, like Kafka.
+
+`PartitionedInput` sources maintain the state of each source and re-build that state during recovery.
+
+## Capture -> Output
+
+Outputs in v0.16 are now stateful, and are now configured with a name for the step:
+
+In v0.15 and before:
+
+``` python doctest:SKIP
+flow.capture(StdOutputConfig())
+```
+
+In v0.16+:
+
+``` python
+from bytewax.dataflow import Dataflow
+from bytewax.connectors.stdio import StdOutput
+
+flow = Dataflow()
+flow.output("out", StdOutput())
+```
+
+## New entrypoint
+
+In Bytewax v0.16, the way that Dataflows are run has been simplified, and most execution functions have been removed.
+
+Similar to other Python frameworks like Flask and FastAPI, Dataflows can now be run using a Datfaflow import string in the <module_name>:<dataflow_variable_name_or_factory_function> format. As an example, given a file named `dataflow.py` with the following contents:
```

Comment:
I forgot, does it default to looking for a `flow` variable? Can you leave off the `:flow`?